File: datalib/src/main/java/nl/acidcats/tumblrlikes/util/prefs/PrefsHelper.java
package nl.acidcats.tumblrlikes.util.prefs;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

/**
 * Created on 04/10/2018.
 */
public class PrefsHelper {

    private SharedPreferences _prefs;

    public PrefsHelper(Context context, @NonNull String name) {
        _prefs = context.getSharedPreferences(name, Context.MODE_PRIVATE);
    }

    public void putString(String key, String value) {
        _prefs.edit().putString(key, value).apply();
    }

    public void putLong(String key, long value) {
        _prefs.edit().putLong(key, value).apply();
    }

    public String getString(String key) {
        return getString(key, null);
    }

    public String getString(String key, @Nullable String defValue) {
        return _prefs.getString(key, defValue);
    }

    public long getLong(String key) {
        return _prefs.getLong(key, 0L);
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/util/prefs/PrefsHelper.kt
package nl.acidcats.tumblrlikes.util.prefs

import android.content.Context

/**
 * Created on 25/10/2018.
 */
class PrefsHelper constructor(context: Context, name: String) {
    private val prefs = context.getSharedPreferences(name, Context.MODE_PRIVATE)

    fun putString(key: String, value: String) = prefs.edit().putString(key, value).apply()

    fun getString(key: String): String? = getString(key, null)

    fun getString(key: String, defValue: String?): String? = prefs.getString(key, defValue)

    fun putLong(key: String, value: Long) = prefs.edit().putLong(key, value).apply()

    fun getLong(key: String): Long = prefs.getLong(key, 0L)
}